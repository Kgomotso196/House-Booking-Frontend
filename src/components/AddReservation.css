.resrevationForm {
  display: flex;
  justify-content: center;
  flex-direction: column;
  gap: 3rem;
}

.formElement {
  display: flex;
  justify-content: space-between;
  gap: 5rem;
  margin-right: 0;
  padding-right: 0;
}
