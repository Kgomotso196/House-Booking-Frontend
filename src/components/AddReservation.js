import React, { useState } from 'react';

const AddReservation = () => {
  const [formData, setFormData] = useState({
    checking_date: '',
    checkout_date: '',
    city: '',
    // Add other fields as per your reservation model
  });

  const handleSubmit = async (event) => {
    event.preventDefault();
    try {
      const response = await fetch('http://3001/api/v1/reservations', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(formData),
      });
      if (response.ok) {
        // Redirecting user
      } else {
        // Handling error
      }
    } catch (error) {
      // Handling error
    }
  };

  const handleChange = (event) => {
    const { name, value } = event.target;
    setFormData({ ...formData, [name]: value });
  };

  return (
    <div>
      <h1>Add Reservation</h1>
      <form onSubmit={handleSubmit}>
        <input type="text" name="checking_date" value={formData.checking_date} onChange={handleChange} placeholder="Checking Date" />
        {/* Reservation details will come here */}
        <button type="submit">Add Reservation</button>
      </form>
    </div>
  );
};

export default AddReservation;
