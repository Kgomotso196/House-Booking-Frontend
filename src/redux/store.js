import { configureStore } from '@reduxjs/toolkit';
import getReservationSlice from './slices/GetReservationSlice';
import getHosesSlice from './slices/GetHosesSlice';
import postReservationSlice from './slices/PostReservationSlice';
import postHousesSlice from './slices/PostHousesSlice';

const store = configureStore({
  reducer: {
    reservations: getReservationSlice,
    houses: getHosesSlice,
    postReservations: postReservationSlice,
    postHouses: postHousesSlice,
  },
});

export default store;
