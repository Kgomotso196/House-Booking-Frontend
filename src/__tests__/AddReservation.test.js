/* eslint-env jest */
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import AddReservation from '../components/AddReservation';

describe('Test House Component', () => {
  test('it renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <AddReservation />
          </Router>
        </Provider>,
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
