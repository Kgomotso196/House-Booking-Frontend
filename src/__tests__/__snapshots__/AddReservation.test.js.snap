// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test House Component it renders correctly 1`] = `
<div
  className="reservationCard"
>
  <h1>
    Add Reservation
  </h1>
  <form
    className="resrevationForm"
    onSubmit={[Function]}
  >
    <div>
      <label
        className="formElement"
        htmlFor="checking_date"
      >
        <div>
          Checking Date:
        </div>
        <input
          className="reservationInput"
          id="checking_date"
          name="checking_date"
          onChange={[Function]}
          placeholder="dd/mm/yyyy"
          type="text"
          value=""
        />
      </label>
    </div>
    <div>
      <label
        className="formElement"
        htmlFor="checkout_date"
      >
        <div>
          Checkingout Date:
        </div>
        <input
          className="reservationInput"
          id="checkout_date"
          name="checkout_date"
          onChange={[Function]}
          placeholder="dd/mm/yyyy"
          type="text"
          value=""
        />
      </label>
    </div>
    <div>
      <label
        className="formElement"
        htmlFor="city"
      >
        <div>
          City:
        </div>
        <input
          className="reservationInput"
          id="city"
          name="city"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
    </div>
    <div>
      <label
        className="formElement"
        htmlFor="house_price"
      >
        <div>
          House Price:
        </div>
        <input
          className="reservationInput"
          id="house_price"
          name="house_price"
          onChange={[Function]}
          type="text"
          value=""
        />
      </label>
    </div>
    <div>
      <label
        className="formElement"
        htmlFor="house"
      >
        Select House
        <select
          name="house"
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          >
            Select a house
          </option>
        </select>
      </label>
    </div>
    <button
      onClick={[Function]}
      type="submit"
    >
      Add Reservation
    </button>
  </form>
</div>
`;
